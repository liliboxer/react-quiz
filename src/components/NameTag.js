import React, { Component } from 'react';
import TextSelector from './TextSelector';
import Shape from './Shape';
import ColorSelector from './ColorSelector';

export default class NameTag extends Component {
  state = {
    text: '',
    color: '',
    backgroundColor: ''
  }

  handleTextChange = ({ target }) => {
    this.setState({ text: target.value });
  }

  handleColorChange = ({ target }) => {
    this.setState({ color: target.value });
  }

  handleBackgroundColorChange = ({ target }) => {
    this.setState({ backgroundColor: target.value });
  }

  handleSubmit = event => {
    event.preventDefault();
    this.setState(state => {
      return ({ 
        text: state.text
      });
    });

  }

  render() {
    return (
      <>
        <TextSelector
          text={this.state.text}
          handleTextChange={this.handleTextChange}
          handleSubmit={this.handleSubmit}
        />

        <Shape
          text={this.state.text}
          color={this.state.color}
          backgroundColor={this.state.backgroundColor}
        />

        <ColorSelector
          color={this.state.color}
          backgroundColor={this.state.backgroundColor}
          handleColorChange={this.handleColorChange}
          handleBackgroundColorChange={this.handleBackgroundColorChange}
        />
      </>
    );
  }
}
