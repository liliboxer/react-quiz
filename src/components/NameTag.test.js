import React from 'react';
import { shallow } from 'enzyme';
import NameTag from './NameTag';

describe('NameTag component', () => {
  it('renders NameTag', () => {
    const wrapper = shallow(<NameTag />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
