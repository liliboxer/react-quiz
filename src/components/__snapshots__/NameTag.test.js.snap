// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameTag component renders NameTag 1`] = `
<Fragment>
  <TextSelector
    handleSubmit={[Function]}
    handleTextChange={[Function]}
    text=""
  />
  <Shape
    backgroundColor=""
    color=""
    text=""
  />
  <ColorSelector
    backgroundColor=""
    color=""
    handleBackgroundColorChange={[Function]}
    handleColorChange={[Function]}
  />
</Fragment>
`;
